xbmc/settings/SettingsBase.h

```diff
@@ -103,9 +103,14 @@ class CSettingsBase
    \brief Gets the setting with the given identifier.
 
    \param id Setting identifier
+   \param[in] notPresentPossible If is set, error messages are ignored because the given setting
+                                 might not actually exist.
+                                 Can refer to background settings related to Kodi, which may be.
+                                 added later, e.g. to an add-on.
    \return Setting object with the given identifier or NULL if the identifier is unknown
    */
-  std::shared_ptr<CSetting> GetSetting(const std::string& id) const;
+  std::shared_ptr<CSetting> GetSetting(const std::string& id,
+                                       bool notPresentPossible = false) const;
```

Comment:
Suggest renaming `notPresentPossible` to `presenceValidated`, and switch the default to true.

Reply:
Thanks @phunkyfish, have changed. This now much better as name :+1: .